src/lib.rs

```diff
@@ -253,6 +257,30 @@ pub async fn start_lemmy_server(args: CmdArgs) -> Result<(), LemmyError> {
   Ok(())
 }
 
+/// Creates temporary HTTP server which returns status 503 for all requests.
+fn create_startup_server(http_server_enabled: bool) -> Result<ServerHandle, LemmyError> {
+  let startup_server = HttpServer::new(move || {
+    App::new().wrap(ErrorHandlers::new().default_handler(move |req| {
+      let (req, _) = req.into_parts();
+      let message = if http_server_enabled {
+        "Lemmy is currently starting"
+      } else {
+        "HTTP server disabled"
+      };
+      let response = HttpResponse::ServiceUnavailable().json(json!({"error": message}));
+      let service_response = ServiceResponse::new(req, response);
+      Ok(ErrorHandlerResponse::Response(
+        service_response.map_into_right_body(),
+      ))
+    }))
+  })
+  .bind((SETTINGS.bind, SETTINGS.port))?
```

Comment:
seems like if http server is disabled this intermediary server should also be disabled?

Reply:
I was thinking to start it anyway in case someone tries to connect and wonders why it fails. Thats why I added a separate error message for that case.

Reply:
ah i see you also keep it running if http server is disabled. That kind of makes sense then but I still don't think it's a good because you're binding the port which prevents any other process from also binding the port. So it prevents you from running `lemmy_server --disable-http-server` next to a normal lemmy_server on the same server

Reply:
Makes sense, I changed it.